javers-persistence-sql/src/main/java/org/javers/repository/sql/finders/CdoSnapshotsEnricher.java

```diff
@@ -0,0 +1,83 @@
+import java.util.*;
+
+public class CdoSnapshotsEnricher {
+
+    private CommitPropertyFinder commitPropertyFinder;
+
+    public CdoSnapshotsEnricher(CommitPropertyFinder commitPropertyFinder) {
+        this.commitPropertyFinder = commitPropertyFinder;
+    }
+
+    List<CdoSnapshot> enrichSnapshots(List<CdoSnapshotDTO> snapshotDTOs) {
+        if (snapshotDTOs.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        Map<Long, Map<String, String>> commitsProperties = prepareCommitsProperties(snapshotDTOs);
+        return enrichSnapshotsWithCommitProperties(snapshotDTOs, commitsProperties);
+    }
+
+    private Map<Long, Map<String, String>> prepareCommitsProperties(List<CdoSnapshotDTO> snapshotDTOs) {
+        List<Long> commitPKs = getCommitPKs(snapshotDTOs);
+        List<CommitPropertyDTO> commitPropertyDTOs = commitPropertyFinder.findPropertiesOfCommits(commitPKs);
+        return convertToMap(commitPropertyDTOs);
+
+    }
+
+    private List<Long> getCommitPKs(List<CdoSnapshotDTO> snapshotDTOs) {
+        List<Long> commitPKs = new ArrayList<>();
+        for (CdoSnapshotDTO snapshotDTO : snapshotDTOs) {
+            if (!commitPKs.contains(snapshotDTO.getCommitPK())) {
+                commitPKs.add(snapshotDTO.getCommitPK());
+            }
+        }
+        return  commitPKs;
+    }
+
+    private Map<Long, Map<String, String>> convertToMap(List<CommitPropertyDTO> commitPropertyDTOs) {
+        Map<Long, Map<String, String>> commitsProperties = new HashMap<>();
+        for (CommitPropertyDTO commitPropertyDTO : commitPropertyDTOs) {
+            if (!commitsProperties.containsKey(commitPropertyDTO.getCommitPK())) {
+                commitsProperties.put(commitPropertyDTO.getCommitPK(), new HashMap<String, String>());
+            }
+            commitsProperties.get(commitPropertyDTO.getCommitPK()).put(commitPropertyDTO.getName(), commitPropertyDTO.getValue());
+        }
+        return commitsProperties;
+    }
+
+    private List<CdoSnapshot> enrichSnapshotsWithCommitProperties(List<CdoSnapshotDTO> snapshotDTOs, final Map<Long, Map<String, String>> commitsProperties) {
+        return Lists.transform(snapshotDTOs, new Function<CdoSnapshotDTO, CdoSnapshot>() {
+            @Override
+            public CdoSnapshot apply(CdoSnapshotDTO cdoSnapshotDTO) {
+                CdoSnapshot snapshot = cdoSnapshotDTO.getSnapshot();
+                long commitPK = cdoSnapshotDTO.getCommitPK();
+                if (commitsProperties.containsKey(commitPK)) {
+                    CommitMetadata commitMetadata = snapshot.getCommitMetadata();
+                    Map<String, String> commitProperties = commitsProperties.get(commitPK);
+                    setPropertiesOnCommitMetadata(commitMetadata, commitProperties);
+                }
+                return snapshot;
+            }
+            private void setPropertiesOnCommitMetadata(CommitMetadata commitMetadata, Map<String, String> commitProperties) {
```

Comment:
to nie wygląda ok, może zrobić CommitMetadataBuilder i konstruować CommitMetadata jak już będą wszystkie dane?